src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js

```diff
@@ -37,10 +38,32 @@ export default ajaxCaller(class LaunchAnalysisModal extends Component {
     }
   }
 
+  componentDidMount() {
+    const { entitySelectionModel: { type, selectedEntities }, processSingle, config: { rootEntityType } } = this.props
+
+    if (!processSingle) {
+      const entities = Utils.switchCase(type,
+        [EntitySelectionType.chooseRows, () => _.keys(selectedEntities)],
+        [EntitySelectionType.processMergedSet, () => _.flow(
+          _.flatMap(`attributes.${rootEntityType}s.items`),
+          _.map('entityName')
+        )(selectedEntities)],
+        [Utils.DEFAULT, () => selectedEntities]
+      )
+
+      this.setState({ entities })
```

Comment:
Now that this calculation is synchronous, can we do it in `render` and eliminate this piece of state?

Reply:
Moved to a `this.` method, since it's also used in the launch method.